is_core/templatetags/forms.py

```diff
@@ -36,17 +36,33 @@ def fieldset_renderer(context, form, fieldset):
     context_dict.update({
         'class': values.get('class'),
     })
-    if inline_view:
-        return inline_view.render(context, fieldset[0])
-    template = values.get('template') or 'is_core/forms/default_fieldset.html'
-    context_dict.update({
-        'fields': values.get('fields'),
-        'form': form,
-        'title': fieldset[0],
-        'class': values.get('class'),
-        'fieldsets': values.get('fieldsets')
-    })
-    return render_to_string(template, context_dict, request=request)
+
+    rendered_inline_view = (
+        inline_view.render(context.new(context_dict), fieldset[0]) if inline_view else ''
+    )
+
+    rendered_fieldsets = []
+    for sub_fieldset in values.get('fieldsets', ()):
+        rendered_fieldset = fieldset_renderer(context, form, sub_fieldset)
+        if rendered_fieldset:
+            rendered_fieldsets.append(rendered_fieldset)
+
+    fields = values.get('fields')
+
+    if fields or rendered_fieldsets or rendered_inline_view:
+        template = values.get('template') or 'is_core/forms/default_fieldset.html'
```

Comment:
how about putting the default templates in settings?

Reply:
yes we can but it was not needed yet